ports/mimxrt10xx/Makefile

```diff
@@ -58,22 +52,59 @@ INC += \
 
 include ../rules.mk
 
-#-------------- Self-update  --------------
-SELF_CFLAGS  = $(CFLAGS) -DTINYUF2_SELF_UPDATE
-SELF_LDFLAGS = $(LDFLAGS)
-SELF_LD_FILES ?= $(PORT_DIR)/linker/$(MCU)_nor.ld $(PORT_DIR)/linker/self_update_memory.ld $(PORT_DIR)/linker/common.ld
-SELF_ASFLAGS = $(ASFLAGS)
+# Flash binary to correct bootloader address
+flash-pyocd-imxrt: $(BUILD)/$(OUTNAME).bin
+	pyocd flash -t $(PYOCD_TARGET) -a 0x60000400 $<
+	pyocd reset -t $(PYOCD_TARGET)
 
-include ../self_update.mk
+#------------- Load to SRAM using sdphost -------------
+# Note: you may need to give the sdphost binary executable permission first.
+#---------------------------------------------------------
+SDP_MIMXRT1011_PID = 0x0145
+SDP_MIMXRT1011_WRITE_ADDR = 0x20206400
+SDP_MIMXRT1011_JUMP_ADDR  = 0x20207000
 
-# required for uf2 conversion
-$(SELF_BUILD_OBJ)/$(SELF_OUTNAME).hex: $(SELF_BUILD_OBJ)/$(SELF_OUTNAME).elf
-	@echo CREATE $@
-	@$(OBJCOPY) -O ihex -R .flash_config -R .ivt $^ $@
+SDP_MIMXRT1021_PID = 0x0130
+SDP_MIMXRT1021_WRITE_ADDR = 0x0400
+SDP_MIMXRT1021_JUMP_ADDR  = 0x1000
+
+SDP_MIMXRT1062_PID = 0x0135
+SDP_MIMXRT1062_WRITE_ADDR = 0x0400
+SDP_MIMXRT1062_JUMP_ADDR  = 0x1000
+
+ifeq ($(OS),Windows_NT)
+	SDPHOST = sdphost/win/sdphost.exe
+else
+	UNAME_S := $(shell uname -s)
+	UNAME_I := $(shell uname -i)
 
-# required for self-update elf compile
-$(SELF_BUILD_OBJ)/self_update/$(OUTNAME).c: $(SELF_BUILD_OBJ)/self_update/$(OUTNAME)-textonly.bin
-	$(PYTHON3) $(TOP)/lib/uf2/utils/uf2conv.py --carray $^ -o $@
+	ifeq ($(UNAME_S),Darwin)
+		SDPHOST = sdphost/mac/sdphost
+	endif
 
-$(SELF_BUILD_OBJ)/self_update/$(OUTNAME)-textonly.bin: $(BUILD)/$(OUTNAME).elf
-	@$(OBJCOPY) -O binary -R .flash_config -R .ivt $^ $@
+	ifeq ($(UNAME_S),Linux)
+		ifeq ($(UNAME_I),x86_64)
+			SDPHOST = sdphost/linux/amd64/sdphost
+	  endif
+
+	  ifeq ($(UNAME_I),i386)
+			SDPHOST = sdphost/linux/i386/sdphost
+	  endif
+	endif
+endif
+
+load-sdp: $(BUILD)/$(OUTNAME).bin

```

Comment:
@nxf58843 this will make life easier for us, though I can only find the sdphost for x86, **do you know how or where to get the sdphost for ARM/linux e.g RPi**. Adafruit preferably use RPi instead of PC for production since it is cheaper, consume less power and basically easier to set up.

Reply:
Compiled versions for linux, mac and windows are included in the tools/sdphost folder of the [IMX-RT1050-FLASHLOADER](https://www.nxp.com/webapp/Download?colCode=IMX-RT1050-FLASHLOADER&appType=license)

To run on the RPi, you would need to build from the source.  The source code is included in the SDK if you include the "MCU Boot" middleware option under SDK_2.8.2_EVK-MIMXRT1010/middleware/mcu-boot/tools/tools/sdphost/sdphost/

Here is a link to the [RT1010 SDK for linux with all middleware](https://mcuxpresso.nxp.com/download/88af2ba86fae2830192edc464d095333)

Reply:
superb !! Thank you very much, I will include the src since it is also BSD License. ( I am too lazy to do crosscompile for RPi for now).